profiler/profiler.go

```diff
@@ -64,15 +62,14 @@ func Stop() {
 // profiler collects and sends preset profiles to the Datadog API at a given frequency
 // using a given configuration.
 type profiler struct {
-	mu         sync.Mutex
-	cfg        *config                      // profile configuration
-	out        chan batch                   // upload queue
-	uploadFunc func(batch) error            // defaults to (*profiler).upload; replaced in tests
-	exit       chan struct{}                // exit signals the profiler to stop; it is closed after stopping
-	stopOnce   sync.Once                    // stopOnce ensures the profiler is stopped exactly once.
-	wg         sync.WaitGroup               // wg waits for all goroutines to exit when stopping.
-	met        *metrics                     // metric collector state
-	prev       map[string]*pprofile.Profile // previous collection results for delta profiling
+	cfg        *config           // profile configuration
+	out        chan batch        // upload queue
+	uploadFunc func(batch) error // defaults to (*profiler).upload; replaced in tests
+	exit       chan struct{}     // exit signals the profiler to stop; it is closed after stopping
+	stopOnce   sync.Once         // stopOnce ensures the profiler is stopped exactly once.
+	wg         sync.WaitGroup    // wg waits for all goroutines to exit when stopping.
+	met        *metrics          // metric collector state
+	deltas     map[string]*deltaProfiler
```

Comment:
`ProfileType` would be better. I'd need to pass the `ProfileType` through `collectGenericProfile` so it can be used to look up the delta profiler. See my comment on another PR where I add that parameter: https://github.com/DataDog/dd-trace-go/pull/1485#discussion_r982435139

Reply:
Now that #1485 is merged, there's a `ProfileType` parameter for `collectGenericProfile` so I rebased off of main and used it.